pkg/kubernetessecrets/kubernetes_secrets_builder.go

```diff
@@ -51,7 +56,7 @@ func CreateKubernetesSecretFromItem(kubeClient kubernetesClient.Client, secretNa
 
 	if currentSecret.Annotations[VersionAnnotation] != itemVersion {
```

Comment:
@SimonBarendse I as thinking of also changing this to 
```suggestion
	for key, value := range secretAnnotations {
		if currentSecret.Annotations[key] != value {
```

That way it'll iterate through the expected secretAnnotations and if any of those on the existing secret are incorrect it'd update them. The same goes I guess for labels. We'd probably need to do the exact same for labels. What do you think?

Reply:
Yes, I think this will be needed to add/change/delete annotations without version bumps.

There's a couple more things we should do I believe:
1. Make the lookup accept new keys:
```go
exists, annotation := currentSecret.Annotations[key]
if !exists || annotation != value
```
2. Also check for annotations on the current secret, that are no longer in the updated secret.

And the same for labels.

Given all that, this equality check becomes quite elaborate. Are there existing equality functions we can use? Or can we just call update every time, which will result in no changes if it's already up to date? @jillianwilson or @florisvdg do you know what would be the impact of calling update without checking for changes first? Would that be okay?

Reply:
@jillianwilson or @florisvdg any thoughts?

Reply:
@SimonBarendse I've added some logic - I'm not sure how we could possibly manage this, however:
> Also check for annotations on the current secret, that are no longer in the updated secret.

My concern is that Kubernetes will automatically add some annotations to the secret and unless we maintain a list of annotations to ignore (I'm thinking things like `kubectl.kubernetes.io/last-applied-configuration`) then we'll go into a vicious loop of updating secrets constantly. 

We're using this code on our clusters running my fork and it's working very well for the last few weeks. Would be awesome if we could get this in properly to the maintained codebase.

Reply:
> My concern is that Kubernetes will automatically add some annotations to the secret and unless we maintain a list of annotations to ignore (I'm thinking things like `kubectl.kubernetes.io/last-applied-configuration`) then we'll go into a vicious loop of updating secrets constantly.

Kubernetes doesn't add these annotations when the secrets are created through the 1Password operator, so this shouldn't be a problem.

> Are there existing equality functions we can use instead?

We can use [`reflect.DeepEqual`](https://pkg.go.dev/reflect#DeepEqual).

> Can we just call update every time, which will result in no changes if it's already up to date?

We can't, as this will trigger a redeploy for all deployments that are using the secret.

---

> Also check for annotations on the current secret, that are no longer in the updated secret.

To elaborate why I'm pushing on this change: if we don't add this check, removals of annotations will not propagate to the secrets. In the example you gave in #41, if you remove the annotations the Jenkins plugin acts on from the OnePasswordItem CRD, these changes won't propagate to the Kubernetes Secret and the credential will remain available in Jenkins, even though the successful apply of that CRD suggests that it isn't available to Jenkins anymore.

Reply:
So I believe all that's needed is replacing the calls to `CompareSecretFieldsWithOnePasswordItem` with calls to `reflect.DeepEqual`.

Reply:
Sounds good, I've done that.